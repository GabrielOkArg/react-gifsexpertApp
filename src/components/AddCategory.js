import React, { useState } from 'react';
import PropTypes from 'prop-types';



export const AddCategory = ({setCategories}) => {
    const [inputValue, setinputValue] = useState('');

    const handleInputValue =(e)=>{
       setinputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        setCategories(cate => [inputValue,...cate]);
        setinputValue('');
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                placeholder='Que gif estas buscando?'
                value={inputValue}
                onChange={handleInputValue}
                required
            />
        </form>
    );
   
}

AddCategory.protoTypes={
    setCategories: PropTypes.func.isRequired
}
