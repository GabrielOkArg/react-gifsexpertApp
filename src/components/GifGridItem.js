import React from 'react';
import PropTypes from 'prop-types'; //impt

//import '../index.css';

export const GifGridItem = ({title,url}) => {

    return (
        <div className='card animate__animated animate__fadeInDown'>
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
        );
};

GifGridItem.protoType={
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired
}
