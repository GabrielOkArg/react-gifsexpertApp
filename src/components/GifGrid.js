import React, { useState, useEffect } from 'react';
import { useFecthGifs } from '../hooks/useFetchGifs';
//import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';


export const GifGrid = ({ category }) => {

    const {data:images,lodading} = useFecthGifs(category);

    return (
        <>
             <h3>{category}</h3>
           {/* el operador && sirve para no hacer un ternario no evalua el false */}
           {lodading && <p>Loading</p>} 
            <div className='card-grid'>
                {
                    images.map((item) => {
                        return <GifGridItem key={item.id} {...item} />
                    })
                }
            </div>
        </>
    );
};
