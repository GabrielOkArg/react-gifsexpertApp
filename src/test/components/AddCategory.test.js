import React from 'react';
import { shallow } from "enzyme"
import { AddCategory } from "../../components/AddCategory"



describe('Pruebas en <AddCategori/>', () => {

    const setCategories=()=>{}
    const wrapper = shallow(<AddCategory setCategories={setCategories}/>)
    test('Debe mostrar correctamente', () => {        
        expect(wrapper).toMatchSnapshot()
    });
    test('Debe cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value ='Hola Mundo';
        //para simular el onChange, necesita tambien tener el evento e y su target y value
        input.simulate('Change',{target:{value:value}});
       expect( wrapper.find('p').text().trim()).toBe(value)
    })    
})