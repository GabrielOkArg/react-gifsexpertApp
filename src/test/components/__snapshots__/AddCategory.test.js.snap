// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategori/> Debe mostrar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Que gif estas buscando?"
    required={true}
    type="text"
    value=""
  />
</form>
`;
